commands/describe_test.go

```diff
@@ -0,0 +1,29 @@
+package commands
+
+import "testing"
+
+func Test_getFunctionURLs(t *testing.T) {
+	cases := []struct {
+		name              string
+		gateway           string
+		functionName      string
+		functionNamespace string
+		expectedURL       string
+		expectedAsyncURL  string
+	}{
+		{"localhost", "http://127.0.0.1:8080", "figlet", "alpha", "http://127.0.0.1:8080/function/figlet.alpha", "http://127.0.0.1:8080/async-function/figlet.alpha"},
+		{"secure site", "https://example.com", "nodeinfo", "beta", "https://example.com/function/nodeinfo.beta", "https://example.com/async-function/nodeinfo.beta"},
+		{"no namespace", "https://example.com:31112", "nodeinfo", "", "https://example.com:31112/function/nodeinfo", "https://example.com:31112/async-function/nodeinfo"},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			url, asyncURL := getFunctionURLs(tc.gateway, tc.functionName, tc.functionNamespace)
+
+			if url != tc.expectedURL || asyncURL != tc.expectedAsyncURL {
+				t.Fatalf("incorrect URL(s):\n got %s and %s\n expected %s and %s\n", url, asyncURL, tc.expectedURL, tc.expectedAsyncURL)
+			}
+			t.Log("Expected URLs were generated: " + url + " and " + asyncURL + ".")
```

Comment:
What is the purpose of this log statement?

Reply:
I was thinking having a log of what went well would distinguish it from a test that didn't have tests and passed. I encountered PASS when I mistakenly had no tests, ran the test, and the result was PASS, because nothing failed.

There's a difference between PASS because there was no failure, and actually PASS because the tests were successful.

However, if it's not useful, not idiomatic, nor per the style of this repository, I'll remove the `t.Log`.

Reply:
I looked at several other tests in faas-cli/commands, and I didn't see any that logged the details of what resulted in a test PASS. So, for consistency, I'll remove the line beginning with `t.Log`.

Reply:
Thanks